### ISSUE TITLE
AOR_Reports: PHP generate  'Illegal string offset' and 'Cannot assign an empty string to a string offset' exceptions

### ISSUE THREAD

**Issue by cherrador** [salesagility/SuiteCRM#7604]
While running an AOR_Report on PHP 7 up, it generate these warning because the erroneous way a string is coerced to handle an array. 

#### Issue
This is warning at the moment, which will be escalated to deprecation in the future. It fills the PHP logs with warnings.

#### Expected Behavior
Code should run without warnings.

#### Actual Behavior
For each field on the AOR_Report an exception warning is raised when the field attribute 'params' is empty. It generate any of these two exceptions depending the situation:

Illegal string offset 'record_id' at AOR_Report.php, line 820
Cannot assign an empty string to a string offset at AOR_Report.php, line 820

#### Possible Fix
Modify file /modules/AOR_Reports/AOR_Report.php at line 820. Preparing the string as array fix the problem from:
```php
$att['params']['record_id'] = $row[$att['alias'] . '_id'];
```
to:
```php
$att['params'] = [];
$att['params']['record_id'] = $row[$att['alias'] . '_id'];
```

#### Steps to Reproduce
1. SuiteCRM installed on PHP 7 or higher.
2. PHP with warnings and logs enabled.
3. Create an AOR_report with fields that do not have specified format.
4. Run report, logs will report the warnings for each field.

#### Context
I'm trying to trace other major bug on the AOR_Reports, but cannot find it with all these noise. I need to correct it before proceeding. These warnings will become deprecations that will raise even more the seriousness level. 

#### Your Environment
* SuiteCRM Version 7.8.30
* Chrome Version 75.0.3770.100 (Official Build) (64-bit)
* PHP 7.1.30
* Windows 2016 Server.

**Comment by Dillon-Brown** [comment]
Fixed here: #6794

**Comment by cherrador** [comment]
@Dillon-Brown I appreciate that you took your time to look on this one. However this report is for 7.8 not 7.10. I double check both versions and I can see that, yes, it was fixed on 7.10 but not ported to the 7.8 branch. I see the fix is different than mine, as I was being conservative avoiding changing the used variables.

Unfortunately we cannot migrate up from 7.8 yet. I am doing my own fixes, but I rather that are included on the main code. So there is no conflict on the the next patch upgrade.
